### ISSUE TITLE
New rule proposal: Attribute definition should not span multiple lines

### ISSUE THREAD

**Issue by vweijsters** [DotNetAnalyzers/StyleCopAnalyzers#810]
**Category:** Layout
**Title:** Attribute definition should not span multiple lines
**Description:** C# allows attributes to be specified across multiple lines of code, like:
```C#
[
Export
]
public class Foo
{
}
```
This form makes it more difficult to understand the attribute definition. Each attribute should therefore be placed on a single line.

```C#
[Export]
public class Foo
{
}
```

**Comment by Przemyslaw-W** [comment]
+1

**Comment by sharwell** [comment]
I don't like the way this one is worded. I prefer to see it defined as a pair of rules (even if both rules are covered by a single diagnostic:

1. The opening `[` of an attribute cannot be the last token on a line.
2. The closing `]` of an attribute cannot be the first token on a line.

**Comment by vweijsters** [comment]
@sharwell That would better fit the style of the current rules indeed. 

My proposal would be to split this issue into two new ones as soon as we have sufficient support for it.

**Comment by sharwell** [comment]
@vweijsters My proposal would be to simply remove the exception from SA1016 (OpeningAttributeBracketsMustBeSpacedCorrectly) and SA1017 (ClosingAttributeBracketsMustBeSpacedCorrectly) which allows this. If we get feedback from users that the exception is actually needed, we can treat it as a bug and open a new issue (or issues) for the functionality described in this proposal.

**Comment by GregReddick** [comment]
I agree with sharwell. Long attributes may wind up spanning multiple lines
```
[SomeAttribute(
   blah1,
   blah2,
   blah3)]
```
I particularly see this with rule suppressions where the justification is too long to make it fit on a line.
